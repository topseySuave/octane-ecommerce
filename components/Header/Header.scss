@import 'colors';
@import 'constants';

.ant-layout-header {
  height: auto !important;
}

.main-header-container {
  position: fixed;
  z-index: 1;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: $headerBackground !important;
  box-shadow: $shadow;

  .main-header {
    display: flex;

    .logo {
      width: 20%;
      height: 31px;
      margin: 16px 28px 16px 0;
      
      svg {
        flex-basis: 55%;
      }
    }

    .icon-cart {
      font-size: $cartFontSize !important;
    }

    .ant-badge {
      position: absolute !important;
      right: 20px !important;
    }
  }

  .mini-header {
    width: 100%;
    display: flex;
    height: (64px / 1.5) !important;
    background: transparent !important;
    justify-content: space-between;
    border-top: 1px solid $headerBorderColor;
  }

  .ant-back-top {
    bottom: 100px;

    .ant-back-top-inner {
      width: max-content;
      padding: 15px;
      line-height: 40px;
      border-radius: 4px;
      background-color: #1088e9;
      color: #fff;
      text-align: center;
    }
  }
}
