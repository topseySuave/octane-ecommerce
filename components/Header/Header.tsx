import { BackTop, Badge, Button, Dropdown, Icon, Layout, Menu } from 'antd';
import AutoComplete from 'components/ui/AutoComplete';
import OctDrawer from 'components/ui/Drawer';
import Logo from 'components/ui/Logo';
import { DISTANCE_FROM_TOP, LINE_HEIGHT } from 'lib/constants';
import routes from 'lib/routes';
import * as React from 'react';
import './Header.scss';
const { Header } = Layout;

const { Link } = routes;

const LayoutHeader: React.SFC = () => {
  const menu = () => (
    <Menu>
      <Menu.Item key="1">
        <Link prefetch href="/signin">
          <a>
            <Icon type="login" className="signin" />
            {'  '}
            Login
          </a>
        </Link>
      </Menu.Item>
      <Menu.Divider />
      <Menu.Item key="4">
        <Link prefetch href="/signup">
          <a>
            <Icon type="user-add" className="signup" />
            {'   '}
            Join Octane and Shop
          </a>
        </Link>
      </Menu.Item>
    </Menu>
  );

  return (
    <Header className="main-header-container">
      <div className="main-header">
        <div className="logo">
          <Link prefetch href="/">
            <a>
              <Logo />
            </a>
          </Link>
        </div>
        <AutoComplete />
        <Menu
          mode="horizontal"
          defaultSelectedKeys={['3']}
          style={{
            backgroundColor: 'transparent',
            lineHeight: `${DISTANCE_FROM_TOP}px`,
          }}
        >
          <Menu.Item key="signin">
            <Dropdown overlay={menu}>
              <a className="ant-dropdown-link" href="#">
                Your Account <Icon type="caret-down" />
              </a>
            </Dropdown>
          </Menu.Item>
          <Menu.Item key="cart">
            <OctDrawer
              ButtonAppearance={({ onClick }: { onClick: () => void }) => {
                return (
                  <Button type="link" onClick={onClick}>
                    <Badge count={5}>
                      <span className="head-example" />
                    </Badge>
                    <Icon type="shopping" className="icon-cart" />
                  </Button>
                );
              }}
            />
          </Menu.Item>
        </Menu>
      </div>
      <Header className="mini-header">
        <Menu
          mode="horizontal"
          defaultSelectedKeys={['3']}
          style={{
            backgroundColor: 'transparent',
            lineHeight: `${LINE_HEIGHT}px`,
          }}
        >
          <Menu.Item key="category 1">French</Menu.Item>
          <Menu.Item key="category 2">Italian</Menu.Item>
          <Menu.Item key="category 3">Irish</Menu.Item>
          <Menu.Item key="category 4">Animal</Menu.Item>
          <Menu.Item key="category 5">Flowers</Menu.Item>
          <Menu.Item key="category 6">Chrismas</Menu.Item>
          <Menu.Item key="category 7">Valentine's</Menu.Item>
        </Menu>
        <Menu
          mode="horizontal"
          defaultSelectedKeys={['3']}
          style={{
            backgroundColor: 'transparent',
            lineHeight: `${LINE_HEIGHT}px`,
          }}
        >
          <Menu.Item key="category 1">Regional</Menu.Item>
          <Menu.Item key="category 2">Nature</Menu.Item>
          <Menu.Item key="category 3">Seasonal</Menu.Item>
        </Menu>
      </Header>
      <BackTop>
        <div className="ant-back-top-inner">
          <Icon type="to-top" /> Back Up
        </div>
      </BackTop>
    </Header>
  );
};

export default LayoutHeader;
