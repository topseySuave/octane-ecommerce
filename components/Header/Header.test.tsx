import { shallow } from 'enzyme';
import React from 'react';

import Header from './Header';

describe('Header', () => {
  it('Renders with default color/size', () => {
    const tree = shallow(<Header />);
    expect(tree).toMatchSnapshot();
    // const mounted = shallow(<Header />);
    // const header = mounted.find('header');
    // console.log(header);
    // expect(header.hasClass('main-header-container')).toBeTruthy();
  });
});
