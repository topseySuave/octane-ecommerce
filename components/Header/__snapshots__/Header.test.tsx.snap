// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders with default color/size 1`] = `ShallowWrapper {}`;
